src/test/java/com/example/EngineService/Domain/EngineTest.java

```diff
@@ -0,0 +1,124 @@
+        EngineType engineType = new EngineType("ABC-123");
+
+        //Act & Assert
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> {new Engine(invalidSerialNumber, engineType);})
+                .withMessageEndingWith("Engine serial number was null or empty or a set of blank spaces. Please provide valid serial number.");
+    }
+
+    @ParameterizedTest
+    @NullSource
+    void constructor_shouldThrowIllegalArgumentException_whenProvidingNullEngineType(EngineType invalidEngineType){
+        //Arrange
+        String serialNumber = "123456789";
+
+        //Act & Assert
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> {new Engine(serialNumber, invalidEngineType);})
+                .withMessageEndingWith("Engine Type was null. Please provide valid engine type.");
+    }
+
+    @Test
+    void constructor_shouldThrowIllegalArgumentException_whenProvidingSerialNumberStartingWithI(){
+        //Arrange
+        String serialNumberWithI = "I-123456789";
+        EngineType engineType= new EngineType("ABC-123");
+
+        //Act & Assert
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> new Engine(serialNumberWithI, engineType))
+                .withMessageEndingWith("Engine serial number can't start with 'I-'. Please provide valid serial number.");
+
+    }
+
+    @Test
+    void setDetails_shouldModifyEngineDetails_whenProvidingValidValue(){
+        //Arrange
+        String serialNumber = "123456789";
+        EngineType engineType = new EngineType("ABC-123");
+        Engine engine = new Engine(serialNumber, engineType);
+
+        String newDetails = String.format("New details for engine %s", serialNumber);
+
+        //Act
+        engine.setDetails(newDetails);
+
+        //Assert
+        assertThat(engine.getDetails()).isEqualTo(newDetails);
+    }
+
+    @Test
+    void setCreationDateTime_shouldModifyEngineCreationDateTime_whenProvidingValidValue(){
+        //Arrange
+        String serialNumber = "123456789";
+        EngineType engineType = new EngineType("ABC-123");
+        Engine engine = new Engine(serialNumber, engineType);
+
+        Instant newCreationDateTime = Instant.now().plusSeconds(60);
+
+
+        //Act
+        engine.setCreationDateTime(newCreationDateTime);
+
+        //Assert
+        assertThat(engine.getCreationDateTime()).isEqualTo(newCreationDateTime);
+    }
+
+    @Test
+    void toString_shouldShowEngineDataToString_whenProvidingValidValues(){
+        //Arrange
+        String serialNumber = "123456789";
+        EngineType engineType = new EngineType("ABC-123");
+        Engine engine = new Engine(serialNumber, engineType);
+
+        //Act
+        String toString = engine.toString();
+
+        //Assert
+        String engineToString =
+                "Engine{" +
+                        "serialNumber='" + serialNumber + '\'' +
```

Comment:
### X

Here you are checking much more than the `toString()`: `engineType.getCode()`, `engine.getDetails()` and `engine.getCreationDateTime()`.

Moreover, this test is not really testing the expected output as the result you are checking (`engineToString`) depends on other calls that may change their output as well...